src/Components/Components/src/Rendering/ComponentState.cs

```diff
@@ -51,6 +52,11 @@ public ComponentState(Renderer renderer, int componentId, IComponent component,
             _hasCascadingParameters = true;
             _hasAnyCascadingParameterSubscriptions = AddCascadingParameterSubscriptions();
         }
+
+        if (_renderer.ComponentMetrics != null && _renderer.ComponentMetrics.IsParametersEnabled)
+        {
+            _componentTypeName = component.GetType().FullName;
```

Comment:
Is this perf related or "convenience" related.

Adding a field to ComponentState is not great, since this increments the cost of each component that gets created (which your app can have thousands of).

Reply:
This is for perf. Because we need that type _componentTypeName on all parameter updates on each component instance. (when this metric is enabled)